src/pages/Home.vue

```diff
@@ -1,21 +1,28 @@
 <template>
   <div>
-    <SRNEntry
+    <IdEntry
       :redirectTo="redirectTo"
       :redirectID="redirectID"
       :purpose="purpose"
       :purposeParams="purposeParams"
+      :textObject="textObject"
+      :inputObject="inputObject"
+      :maxNumberOfInput="maxNumberOfInput"
+      :dataSourceObject="dataSourceObject"
+      :program="program"
+      :authType="authType"
     />
   </div>
 </template>
 
 <script>
-import SRNEntry from "@/components/SRNEntry.vue";
+import IdEntry from "@/components/IdEntry.vue";
+import firebaseAPI from "@/services/API/getProgramData.js";
```

Comment:
Wouldn't it be better to use a more functionality-specific name that is independent of what we are using on the backend?
```suggestion
import ProgramAPIService from "@/services/API/getProgramData.js";
```

Reply:
Yes agreed.
Two things to notice here
- A file is called `getProgramData`. There's a "get" in front of the name. That usually signifies a getter or a method. A file should be named by what it contains. This file contains a method that retrieves program data using an API call. So maybe we should think about renaming this file. Just program data is also fine! Because right now it contains code to GET the program data. Maybe we might decide to PUSH/UPDATE/DELETE some program data in the future. Then it won't make sense to create separate files for that code right? All of that code can come under a file called `ProgramData.js`.

- And as Aman suggested, the name should be changed. Again, the fact that it's a **_firebase_** API is something that can be changed in the future. Also, there's no need for this component (Home.vue) to know what kind of cloud provider or backend are we talking to. It just needs to know what a particular method will do.

--------

- Just sharing below what we did in Plio and why we follow that pattern everywhere. It made our lives so much easier/
- In Plio, we would have lots of different types of actions being taken on a Question object in the backend. So we made a file called `Question.js`. In that file we wrote something like
```js
export default {
    getQuestion() { ... },
    deleteQuestion() { ... },
    updateQuestion() { ... }
}
```
- What this code above is doing is exporting an _**object**_ that _**contains**_ these three methods. It's not exporting the methods directly. There's a difference. So while importing this in the parent file, we would write something like `import QuestionAPIService from "../../Question.js"`
- This `QuestionAPIService` is an _**object**_ whose keys are those methods you can see in the code above. And that is why to access a method, we would have to write `QuestionAPIService.updateQuestion(...)`